docs/external-node.md

```diff
@@ -0,0 +1,305 @@
+- The `name` is generated according to the following principles:
+    - Use the `ExternalNode` name directly, if there is only one interface, and interface name is not specified.
+    - Use the format `$ExternalNode.name-$hash($interface.name)[:5]` for other cases.
+- The `externalNode` field is set with the `ExternalNode` name.
+- The `owner` is referring to the `ExternalNode` resource.
+- All labels added on `ExternalNode` are copied to the `ExternalEntity`.
+- One `Endpoint` is created per IP in the interface, and every Endpoint's name uses the interface name if it is set.
+
+A sample of the `ExternalEntity` created for the above `ExternalNode` interface is,
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: ExternalEntity
+metadata:
+  labels:
+    role: db
+  name: vm1
+  namespace: vm-ns
+  ownerReferences:
+    - apiVersion: v1alpha1
+      kind: ExternalNode
+      name: vm1
+      uid: 99b09671-72da-4c64-be93-17185e9781a5
+  resourceVersion: "5513"
+  uid: 5f360f32-7806-4d2d-9f36-80ce7db8de10
+spec:
+  endpoints:
+    - ip: 172.16.100.3
+  externalNode: vm1
+```
+
+Note: **Only one interface is supported for Antrea v1.8**.
+
+## Apply Antrea NetworkPolicy to ExternalNode
+
+Antrea NetworkPolicy applies to an `ExternalNode` by setting the `appliedTo` field with a `externalEntitySelector`.
+The `ExternalNode` object is converted to
+`ExternalEntities`, which are used by `antrea-controller` to calculate the NetworkPolicy span, and `antrea-agent`
+running on the `ExternalNode` receives the internal Antrea NetworkPolicy objects.
+
+Following types of remote addresses are supported when applying Antrea NetworkPolicy to an external Node:
+
+- ExternalNode with a `externalEntitySelector`
+- CIDR with an `ipBlock`
+- fqdn address in an egress rule
+
+Following actions are supported when applying Antrea NetworkPolicy to an external Node:
+
+- Allow
+- Drop
+- Reject
+
+Following is an example to apply Antrea NetworkPolicy to the `ExternalNode` objects labeled with `role=db` to reject the
+SSH connections from IP "172.16.100.5" or from other external Nodes labeled with `role=front`:
+
+```yaml
+kind: NetworkPolicy
+metadata:
+  name: anp1
+  namespace: vm-ns
+spec:
+  priority: 9000.0
+  appliedTo:
+    - externalEntitySelector:
+        matchLabels:
+          role: db
+  egress:
+    - action: Reject
+      ports:
+        - protocol: TCP
+          port: 22
+      to:
+        - externalEntitySelector:
+            matchLabels:
+              role: front
+        - ipBlock:
+            cidr: 172.16.100.5/32
+```
+
+`antrea-agent` running on the target external Node sets up a connection with
+`antrea-controller` using the `ExternalNode` name as an option to receive the internal Antrea NetworkPolicy objects.
+More details please refer to the
+[installation guide](vm-installation.md) for `antrea-agent` on an external Node.
+
+`antrea-agent` realizes the Antrea NetworkPolicy on OVS using OpenFlow entries.
+
+## Security
```

Comment:
Should we name it "RBAC for antrea-agent"?

Reply:
renamed.